apps/angular/src/app/features/anime-table/anime-table.component.ts

```diff
@@ -46,6 +109,138 @@ export class AnimeTableComponent {
+
+	private getSubjectValue<T>(subject$: BehaviorSubject<T>): T | null {
+		let subjectValue: T | null = null;
+		subject$.pipe(take(1)).subscribe(value => {
+			subjectValue = value;
+		});
+		return subjectValue;
+	}
+
+	private subscribeToControls(): void {
+		this.paginator.page.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(() => {
+			this.pageNumber$.next(this.paginator.pageIndex);
+			this.pageSize$.next(this.paginator.pageSize);
+		});
+
+		this.sort.sortChange
+			.pipe(takeUntilDestroyed(this.destroyRef))
+			.subscribe(() => this.ordering$.next(SortParamsMapper.toDto(this.sort.active as SortParams, this.sort.direction)));
+	}
+
+	/** @inheritdoc */
+	public ngOnInit(): void {
+		this.getQueryParams();
+	}
+
+	/** @inheritdoc */
+	public ngOnChanges(changes: SimpleChanges): void {
+		const previousSearchValue = changes['search']?.previousValue;
+		const currentSearchValue = changes['search']?.currentValue;
+		if (previousSearchValue !== currentSearchValue) {
+			this.search$.next(currentSearchValue);
+			this.pageNumber$.next(0);
+		}
+		const previousFilterValue = changes['filter']?.previousValue;
+		const currentFilterValue = changes['filter']?.currentValue;
+		if (this.isFiltersChange(previousFilterValue, currentFilterValue)) {
+			this.filter$.next(currentFilterValue);
+			this.pageNumber$.next(0);
+		}
+	}
+
+	private isFiltersChange(previous: string[] | null, current: string[] | null): boolean {
+		if (previous == null || current == null || previous.length !== current.length) {
+			return true;
+		}
+		for (let i = 0; i < current.length; i++) {
+			if (current[i] !== previous[i]) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/** @inheritdoc */
+	public ngAfterViewInit(): void {
+		this.paginator.pageSize = this.getSubjectValue(this.pageSize$) ?? INITIAL_PAGE_SIZE;
+		this.paginator.pageIndex = this.getSubjectValue(this.pageNumber$) ?? 0;
+
+		this.subscribeToControls();
+
+		this.animeList$ = merge(this.sort.sortChange, this.paginator.page, this.search$, this.filter$)
+			.pipe(
+				debounceTime(DEBOUNCE_TIME),
+				startWith(null),
+				tap(() => {
+					this.searchValueEmitter.emit(this.search);
+					this.filterValueEmitter.emit(this.filter);
+				}),
+				switchMap(() =>
+					this.getAnimeList({
+						pageSize: this.getSubjectValue(this.pageSize$) ?? INITIAL_PAGE_SIZE,
+						pageNumber: this.getSubjectValue(this.pageNumber$) ?? 0,
+						ordering: this.getSubjectValue(this.ordering$) ?? undefined,
+						search: this.search.length > 0 ? this.search : undefined,
+						types: this.filter,
+					}).pipe(catchError((error: unknown) => {
+						this.hasFetchingError$.next(true);
+						return throwError(error);
+					}))),
+				tap(value => {
+					if (value != null) {
+						this.totalCount = value.count;
+						this.paginator.pageIndex = this.getSubjectValue(this.pageNumber$) ?? 0;
+					}
+				}),
+				map(value => {
+					if (value == null) {
+						return [];
+					}
+					return value.results;
+				}),
```

Comment:
When we remove 'startWith(null)' we don't need to check value for 'null' here:
```suggestion
				tap(value => {
					this.totalCount = value.count;
					this.paginator.pageIndex = this.getSubjectValue(this.pageNumber$) ?? 0;

				}),
				map(value => value.results),
```